### PR TITLE
(1388) Search screen filters

### PR DESCRIPTION
**Pull request by richpjames**
# Context
Matchers not only need to search by the characteristics such as location, start date etc added in #673 but also by the characteristics of the AP by using filters.

# Changes in this PR
1. We have to hardcode the service name into requests to the bed search endpoints
2. Some general refactoring
3. We make some changes to the controller to allow for using the search filters
4. We edit and add some util functions for helping us create the view
5. We add the view 

## Screenshots of UI changes

Placement Requests -- shows a list of placementRequests (2)

### After

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Have all changes to any dependencies been deployed to both `preprod` and
    `production` in a backwards compatible way? (This includes our API,
    infrastructure, third-party integrations and any secrets or environment variables)
- [ ] Has a required data migration been included in this change or been done in
    advance?

## Post merge checklist

Once we've merged we now need to release this through to production before
considering it done.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-temporary-accommodation-ui)
and work through the following steps:

- [ ] Has the product manager asked to provide approval for this feature?
  - [ ] Has the product manager approved?
- [ ] Manually approve release to preprod
- [ ] Verify change released to preprod
- [ ] Manually approve release to prod
- [ ] Verify change released to production

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

File: server/views/match/search.njk [hunk 1]
```diff
@@ -2,146 +2,173 @@
...
+				<input type="hidden" name="crn" value="{{crn}}"/>
+				<input type="hidden" name="startDate" value="{{startDate}}"/>
+				<input type="hidden" name="applicationId" value="{{applicationId}}"/>
+				<input type="hidden" name="assessmentId" value="{{assessmentId}}"/>
+				{% for requiredCharacteristic in requiredCharacteristics %}
+					<input type="hidden" name="requiredCharacteristics[]" value="{{requiredCharacteristic}}"/>
+					{%endfor%}
+
+					<div class="govuk-grid-column-one-third">
+						{{govukCheckboxes({
```
Comment (pezholio) [inline server/views/match/search.njk:83]: Could we use [Smaller Checkboxes](https://design-system.service.gov.uk/components/checkboxes/#smaller-checkboxes) here?

File: server/utils/matchUtils.ts [hunk 2]
```diff
@@ -19,15 +19,21 @@ export const mapApiParamsForUi = (apiParams: BedSearchParameters): BedSearchPara
   maxDistanceMiles: apiParams.maxDistanceMiles.toString(),
 })
 
-export const mapApiCharacteristicForUi = (characteristic: CharacteristicPair) => {
-  const result = characteristic.name.startsWith('is') ? characteristic.name.slice(2) : characteristic.name
+export const mapApiCharacteristicForUi = (characteristic: string) => {
```
Comment (pezholio) [inline server/utils/matchUtils.ts:22]: This naming is a bit confusing, as we're not accepting an array and using `map` - how about `translateApiCharacteristicForUi`?

**Review by pezholio — APPROVED** [review]
This is ace thanks! 🎖️ Good to go other than the checkboxes and that minor naming point 👍

File: server/utils/matchUtils.ts (hunk 2 above)
Comment (richpjames) [reply server/utils/matchUtils.ts]: Yeah that's fair

**Pull request merged**